File: fix-log-api/Domain/Entities/Expense.cs
namespace fix_log_api.Domain.Entities
{
    public class Expense
    {
        public int Id { get; set; }

        public required int UserId { get; set; }

        public required string Title { get; set; }

        public required string Details { get; set; }

        public required float Price { get; set; }

        public required int Quantity { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
    }
}

File: fix_log/lib/domain/models/expense.dart
class Expense {
  final int id;
  final String title;
  final String details;
  final double price;
  final int quantity;

  Expense({
    required this.id,
    required this.title,
    required this.details,
    required this.price,
    required this.quantity,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'] as int,
      title: json['title'] as String,
      details: json['details'] as String,
      price: (json['price'] as num).toDouble(),
      quantity: json['quantity'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'details': details,
      'price': price,
      'quantity': quantity,
    };
  }
}
